include/boost/math/statistics/univariate_statistics.hpp

```diff
@@ -512,6 +512,79 @@ inline auto interquartile_range(RandomAccessContainer & v)
     return interquartile_range(v.begin(), v.end());
 }
 
+template<class ForwardIterator>
+auto sorted_mode(const ForwardIterator first, const ForwardIterator last) -> std::vector<typename std::iterator_traits<ForwardIterator>::value_type>
+{
+    using Z = typename std::iterator_traits<ForwardIterator>::value_type;
+    static_assert(std::is_integral<Z>::value, "Floating point values have not yet been implemented.");
+    using Size = typename std::iterator_traits<ForwardIterator>::difference_type;
+
+    std::vector<Z> modes {};
+    modes.reserve(16);
+    Size max_counter {0};
+    Z previous_search {std::numeric_limits<Z>::max()};
+
+    if(first == last)
+    {
+        return modes;
+    }
+    
+    for(auto it {first}; it != last; ++it)
+    {
+        if(*it == previous_search)
+        {
+            continue;
+        }
+        
+        Size current_count {0};
+        auto end_it {it};
+        while(end_it != last && *end_it == *it)
+        {
+            ++current_count;
+            ++end_it;
+        }
+
+        if(current_count > max_counter)
+        {
+            modes.resize(1);
+            modes[0] = *it;
+            max_counter = current_count;
+        }
+
+        else if(current_count == max_counter)
+        {
+            modes.emplace_back(*it);
+        }
+
+        previous_search = *it;
```

Comment:
Instead of using `previous_search`, do `it = end_it`. But actually, take it a step further: change the loop to `while (first != last)` and keep moving `first` forward with `first = end_it`.